libs/blocks/pdf-viewer/pdf-viewer.js

```diff
@@ -0,0 +1,50 @@
+import { getConfig } from '../../utils/utils.js';
+
+const API_SOURCE_URL = 'https://documentcloud.adobe.com/view-sdk/viewer.js';
+const PDF_RENDER_DIV_ID = 'adobe-dc-view';
+const {
+  env,
+  pdfViewerClientIdStage,
+  pdfViewerClientIdProd,
+  pdfViewerReportSuiteQA,
+  pdfViewerReportSuiteProd,
+} = getConfig();
+
+const init = (el) => {
+  const isProdEnv = env.name === 'prod';
+  const children = el?.querySelectorAll(':scope > div');
+  const loadApiScriptTag = document.createElement('script');
+  const url = children[0]?.querySelector('a[href$="pdf"]').href;
+  const fileName = url?.split('/').pop();
+
+  children[0]?.remove();
+
+  if (!url) return;
+
+  loadApiScriptTag.setAttribute('src', API_SOURCE_URL);
+  el?.appendChild(loadApiScriptTag);
+
+  const pdfViewerDiv = document.createElement('div');
+
+  pdfViewerDiv.setAttribute('id', PDF_RENDER_DIV_ID);
+  pdfViewerDiv.classList.add('pdf-container');
+  el?.appendChild(pdfViewerDiv);
+
+  document.addEventListener('adobe_dc_view_sdk.ready', () => {
+    const adobeDCView = new AdobeDC.View(
+      {
+        clientId: isProdEnv ? pdfViewerClientIdProd : pdfViewerClientIdStage,
+        divId: PDF_RENDER_DIV_ID,
+        reportSuiteId: isProdEnv ? pdfViewerReportSuiteProd : pdfViewerReportSuiteQA,
+      },
+    );
+    adobeDCView.previewFile(
+      {
+        content: { location: { url } },
+        metaData: { fileName },
+      },
+    );
+  });
```

Comment:
It's best practice to remove listeners if you can. If you change the listener to a named function, then you can remove it when you're done

```
const handleSdkReady = () => {
  // Your code here
  document.removeEventListener('adobe_dc_view_sdk.ready', handleSdkReady);
};
document.addEventListener('adobe_dc_view_sdk.ready', handleSdkReady);
```